gst/nnstreamer/nnstreamer_plugin_api_impl.c

```diff
@@ -1244,6 +1244,35 @@ gst_tensor_pad_caps_is_flexible (GstPad * pad)
   return ret;
 }
 
+/**
+ * @brief Check current pad caps is sparse tensor.
+ * @param pad GstPad to check current caps
+ * @return TRUE if pad has sparse tensor caps.
+ */
+gboolean
+gst_tensor_pad_caps_is_sparse (GstPad * pad)
```

Comment:
Discussion: refactoring functions for pad caps
How about implementing new function and removing gst_tensor_pad_caps_is_flexible() / gst_tensor_pad_caps_is_sparse() ?
```
/**
 * @brief Get tensor format of current pad caps.
 * @param pad GstPad to check current caps
 * @return The tensor format of current pad caps. _NNS_TENSOR_FORMAT_END if pad does not have tensor caps.
 */
tensor_format gst_pad_get_tensor_format (GstPad * pad);
```

Reply:
okay, Thank you.

Reply:
IMO, after implementing the function, we can add simple macro :)
```
#define gst_tensor_pad_caps_is_static(p) (gst_tensor_pad_get_format (p) == _NNS_TENSOR_FORMAT_STATIC)
#define gst_tensor_pad_caps_is_flexible(p) (gst_tensor_pad_get_format (p) == _NNS_TENSOR_FORMAT_FLEXIBLE)
#define gst_tensor_pad_caps_is_sparse(p) (gst_tensor_pad_get_format (p) == _NNS_TENSOR_FORMAT_SPARSE)
```

Reply:
Thank you for your kind explanation. :+1: